sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1529,9 +1529,11 @@ pi_result piPlatformsGet(pi_uint32 NumEntries, pi_platform *Platforms,
     PrintPiTrace = true;
   }
 
-  if (ZeDebug & ZE_DEBUG_CALL_COUNT) {
-    ZeCallCount = new std::map<const char *, int>;
-  }
+  static std::once_flag ZeCallCountInitialized;
```

Comment:
I wonder if having this as a non-global static is really thread-safe.

Reply:
std::call_once() is thread-safe.
https://en.cppreference.com/w/cpp/thread/call_once
```
Executes the Callable object f exactly once, even if called concurrently, from several threads.
```
You can see the example in the above link to see why we don't need any extra lock for std::once_flag.

Reply:
The "std::once_flag" essentially serves as a mutex for the "std::call_once", and function-scope static variables aren't generally thread-safe.

Reply:
Do you want to change it to a global static variable?
I have seen the case the std::once_flag is used as a function static variable.
https://github.com/intel/llvm/blob/sycl/sycl/source/detail/pi.cpp#L373-L376
Please be more specific on how to address your request.

Reply:
I've hoped you'd convince me that it is safe to have this in this scope. I'd say let's have it here for now.